integration/class-surge.php

```diff
@@ -0,0 +1,184 @@
+		if ( preg_match( self::$cache_config_pattern, $config ) ) {
+			return;
+		}
+
+		// Add a WP_CACHE_CONFIG to wp-config.php.
+		$anchor = "/* That's all, stop editing!";
+		if ( false !== \strpos( $config, $anchor ) ) {
+			$config = \str_replace( $anchor, self::$cache_config . "\n\n" . $anchor, $config );
+		} elseif ( false !== \strpos( $config, '<?php' ) ) {
+			$config = \str_replace( '<?php', "<?php\n\n" . self::$cache_config . "\n", $config );
+		}
+
+		$wp_filesystem->put_contents( $file, $config, FS_CHMOD_FILE );
+	}
+
+	/**
+	 * Remove the Surge cache config.
+	 */
+	public static function remove_cache_config() {
+		$file = self::get_config_file_path();
+
+		if ( ! \wp_is_writable( $file ) ) {
+			return;
+		}
+
+		global $wp_filesystem;
+		\WP_Filesystem();
+
+		$config = $wp_filesystem->get_contents( $file );
+
+		// Remove the define line.
+		$config = preg_replace( self::$cache_config_pattern, '', $config );
+
+		$wp_filesystem->put_contents( $file, $config, FS_CHMOD_FILE );
+	}
+
+	/**
+	 * Get the config file.
+	 *
+	 * @return string|false The config file or false.
+	 */
+	public static function get_config_file_path() {
+		$config_file = false;
+
+		// phpcs:ignore
+		if ( @file_exists( ABSPATH . 'wp-config.php' ) ) {
+
+			/** The config file resides in ABSPATH */
+			$config_file = ABSPATH . 'wp-config.php';
+		// phpcs:ignore
+		} elseif ( @file_exists( dirname( ABSPATH ) . '/wp-config.php' ) && ! @file_exists( dirname( ABSPATH ) . '/wp-settings.php' ) ) {
+
+			/** The config file resides one level above ABSPATH but is not part of another installation */
+			$config_file = dirname( ABSPATH ) . '/wp-config.php';
+		}
+
+		return \apply_filters( 'activitypub_surge_cache_config_file', $config_file );
+	}
+
+	/**
+	 * Maybe add the Surge cache config to the site health.
+	 *
+	 * @param array $tests The site health tests.
+	 *
+	 * @return array The site health tests with the Surge cache config test.
+	 */
+	public static function maybe_add_site_health( $tests ) {
+		if ( ! \is_plugin_active( 'surge/surge.php' ) ) {
+			return $tests;
+		}
+
+		$tests['direct']['activitypub_test_surge_integration'] = array(
+			'label' => \__( 'Surge Test', 'activitypub' ),
+			'test'  => array( self::class, 'test_surge_integration' ),
+		);
+
+		return $tests;
+	}
+
+	/**
+	 * Surge integration test.
+	 *
+	 * @return array The test result.
+	 */
+	public static function test_surge_integration() {
+		$result = array(
+			'label'       => \__( 'Compatibility with Surge', 'activitypub' ),
+			'status'      => 'good',
+			'badge'       => array(
+				'label' => \__( 'ActivityPub', 'activitypub' ),
+				'color' => 'green',
+			),
+			'description' => \sprintf(
+				'<p>%s</p>',
+				\__( 'Surge is well configured to work with ActivityPub.', 'activitypub' )
+			),
+			'actions'     => '',
+			'test'        => 'test_surge_integration',
+		);
+
+		if ( ! \defined( 'WP_CACHE_CONFIG' ) ) {
+			$result['status']         = 'critical';
```

Comment:
With the checklist coming together now, should this be an issue that blocks completion of the Site Health step?

Reply:
It can mess up a lot, but I can downgrade it to a suggestion.